pmd-java/src/test/resources/net/sourceforge/pmd/ast/FullTypeAnnotations.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+import java.awt.Button;
+import java.io.File;
+import java.io.FileFilter;
+import java.security.PrivilegedAction;
+import java.util.Comparator;
+import java.util.concurrent.Callable;
+import java.util.function.Function;
+import java.util.function.IntFunction;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * Type annotation syntax.
+ *
+ * See https://checkerframework.org/jsr308/java-annotation-design.html
+ * for precise spec including BNF.
+ *
+ * https://checkerframework.org/jsr308/java-annotation-design.html#type-names
+ * is particularly interesting
+ */
+public class ParserCornerCases18 {
```

Comment:
We should probably rename this class to avoid confusion :smile: 

Do we have annotated method declarations covered? Like

```java
@Override
public void run() { }
```